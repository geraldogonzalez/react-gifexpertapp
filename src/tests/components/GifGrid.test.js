import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas sobre <GifGrid />', () => {

    test('Debe mostrar el componente correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        
        const wrapper = shallow( <GifGrid category="Dragon Ball" /> );

        expect( wrapper ).toMatchSnapshot();
        
    })

    test('Debe mostrar items cuando se cargan las imÃ¡genes luego de la llamada de useFetchGifs', () => {
        
        const gifs = [{
            id: 0,
            url: 'https://imagen.jpg',
            title: 'Imagen'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        
        const wrapper = shallow( <GifGrid category="Dragon Ball" /> );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

    })
    
    

})
