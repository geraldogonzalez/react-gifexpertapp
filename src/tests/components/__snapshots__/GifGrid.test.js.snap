// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <GifGrid /> Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h2>
    Dragon Ball
  </h2>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas sobre <GifGrid /> Debe mostrar items cuando se cargan las im√°genes luego de la llamada de useFetchGifs 1`] = `
<Fragment>
  <h2>
    Dragon Ball
  </h2>
  <div
    className="card-grid"
  >
    <GifGridItem
      id={0}
      key="0"
      title="Imagen"
      url="https://imagen.jpg"
    />
  </div>
</Fragment>
`;
